src/mecab/src/dictionary.cpp

```diff
@@ -5,6 +5,7 @@
 //  Copyright(C) 2004-2006 Nippon Telegraph and Telephone Corporation
 #include <fstream>
 #include <climits>
+#include <functional>
```

Comment:
```suggestion
// patch for https://github.com/VOICEVOX/open_jtalk/issue/10
// ref: https://learn.microsoft.com/en-us/answers/questions/1348183/std-binary-function-is-missing-in-msvc-14-37-32822
#ifdef _WIN32
#include <functional>
#endif
```
MSVC固有の問題であれば、`#ifdef`を用いてWindowsのみ(どうやらMSVCのみというのは難しそうだったので)読み込むようにしてもいいかもと思いました。(全然mustではなくて、他の環境への影響は少ないほうがいいかなと思った次第です)
また、このファイル(もっと言えばリポジトリ全体)はVOICEVOXとして用意したコードではなく、OpenJTalkのコードであり、オリジナルとの差分がわかりやすい方がいい(何かしらの問題に当たったときに、VOICEVOX側の問題なのか、OpenJTalk側の問題なのかをわかりやすくした方がいい)と思うので、コメントを書いておくと後でわかりやすいかもです...!

Reply:
そもそもこのコードは`functional`に依存している部分があるのでincludeしないといけないかもです。
他の場所で`functional`がincludeされていたから偶然使えていた、みたいな･･･？
https://en.cppreference.com/w/cpp/utility/functional/binary_function
（そういう言及をどこかで見たのを思い出したのですがはずなのですが、どこで見たかわからず。。。。。）

そういえば @sevenc-nanashi さんもLinuxでエラーが出たとおっしゃってましたが、ここを変えると動いた感じでしたか･･･？ 👀

Reply:
ちょっと今`#include <functional>`で検索をかけた感じだと`src\mecab\src\learner_tagger.cpp`にincludeされているだけで、さすがにヘッダーには含まれてませんでした。そして、Linux環境（Ubuntu Focal gcc9.4.0）で#include <functional>`をすべて削除してもビルドできたことから、僕の知識不足かもしれませんが、以前のMSVCやgccなどはは暗黙的になぜか`functional`を読み込んでいて、ビルドができていたと考えます。

Reply:
なるほど...であれば`#ifdef _WIN32 ~~ #endif`はなくても良さそうですね...!
となると、コメントも特になく現在の変更のままでもいいかもですね...

Reply:
ぶっちゃけ正解がわからんって感じです… 😇
えいやでマージしてみる感じで良いかなと…！

Reply:
GitHub Actionsと同じコマンドで、Linux環境（Ubuntu Focal gcc9.4.0）を使って確認してみた感じ、このPR前の共有ライブラリ（`libopenjtalk.so.1.10`）のサイズは2152kB、このPRのサイズも同じく2152kBで変わりませんでした。ただハッシュ値は変わっていました。なので完全に同じという感じではないですが、影響は少ないと思います。